src/components/checkbox/Checkbox.js

```diff
@@ -182,6 +183,7 @@ export default class CheckBoxComponent extends Field {
   }
 
   updateValue(value, flags) {
+    this.setChecked();
```

Comment:
Here you are setting the checked state before the value is updated... you may want to do this after the ```super.updateValue``` call.